Docs: Document Go tracer support (#5898)
github/codeql: docs/codeql/codeql-cli/creating-codeql-databases.rst

.. pull-quote:: Creating databases for Go
  
   For Go, install the Go toolchain (version 1.11 or later) and, if there
   are dependencies, the appropriate dependency manager (such as `dep
   <https://golang.github.io/dep/>`__).
   
   The Go autobuilder attempts to automatically detect Go code in a repository,
   and only runs build scripts in an attempt to fetch dependencies. To force
   CodeQL to use your build script, set the environment variable

>    `CODEQL_EXTRACTOR_GO_BUILD_TRACING=on` or pass a command.

Reviewer: Maybe it's just me, but I was a tiny bit confused by "or pass a command" when first reading this.
Which "command" line argument are you referring to? Why is the command I should pass not named here ("a command"?
But then I realized that you meant `--command`.

Feel free to ignore all of my three suggestions, if it's just me who is confused ^^

```suggestion
   `CODEQL_EXTRACTOR_GO_BUILD_TRACING=on` or use the ``--command`` option.
```

```suggestion
   `CODEQL_EXTRACTOR_GO_BUILD_TRACING=on` or pass a command using the ``--command`` option.
```

```suggestion
   `CODEQL_EXTRACTOR_GO_BUILD_TRACING=on`. Build tracing is also enabled when using ``codeql`` with the ``--command`` option.
```

Author: I think it's always good to be more clear and your first suggestion looks great to me. ✨  Thanks for catching that confusing language.

Reviewer 2: - [ ]

>    CodeQL to use your build script, set the environment variable
>    `CODEQL_EXTRACTOR_GO_BUILD_TRACING=on` or use the ``--command`` option.

Reviewer 3: ```suggestion
   CodeQL to limit extraction to the files compiled by your build script, set the environment variable
   `CODEQL_EXTRACTOR_GO_BUILD_TRACING=on` or use the ``--command`` option to specify a
   build command.
```

.. pull-quote:: Creating databases for Go
  
   For Go, install the Go toolchain (version 1.11 or later) and, if there
   are dependencies, the appropriate dependency manager (such as `dep
   <https://golang.github.io/dep/>`__).
   

>    The Go autobuilder attempts to automatically detect Go code in a repository,

Reviewer 4: ```suggestion
   The Go autobuilder attempts to automatically detect code written in Go in a repository,
```

     codeql database create csharp-database --language=csharp --command='dotnet build /p:UseSharedCompilation=false /t:rebuild'

> - Go project built using a custom build script::

Reviewer 3: This might not be hugely clear because both mechanisms are used at once. Perhaps provide two examples, one with a custom script (and therefore tracing enabled), and one using the default build command and using the environment variable to enable tracing (cross-check @sauyon that I haven't mixed up the logic here)

Author: This makes sense. I've added two examples; one with the `CODEQL_EXTRACTOR_GO_BUILD_TRACING=on` environment variable and one with a custom script.
